test/system/annotator_page_test.rb

```diff
@@ -0,0 +1,101 @@
+    WebMock.allow_net_connect!
+    def setup
+        WebMock.disable!
+        @apikey = LinkedData::Client.settings.apikey
+        @annotator_api = $ANNOTATOR_URL
+        @host = "#{@annotator_api.split('/')[-2]}:443"
+        @annotator_text_area = ".annotator-page-text-area > textarea"
+        @sample_response = fixtures(:annotator)["sample_response"]
+    end
+
+    test "go to annotator page and check if all the inputs and filters are there" do
+        visit root_url
+        click_link(href: '/annotator')
+        assert_selector @annotator_text_area
+        assert_selector 'div.insert-sample-text-button'
+        # Check if there are the annotator's options
+        assert_selector 'label[for="chips-whole_word_only-check"]'
+        assert_selector 'label[for="chips-longest_only-check"]'
+        assert_selector 'label[for="chips-expand_mappings-check"]'
+        assert_selector 'label[for="chips-exclude_numbers-check"]'
+        assert_selector 'label[for="chips-exclude_synonyms-check"]'
+        assert_selector 'div.select-ontologies'
+
+        # Open the advanced options
+        find('div.advanced-options-button').click
+
+        # Check if there are the advanced options
+        assert_selector '#umls_semantic_types_selector'
+        assert_selector '#umls_semantic_groups_selector'
+        assert_selector '#include_accentors'
+        assert_selector '#include_score'
+        assert_selector 'input[name="score_threshold"]'
+        assert_selector 'input[name="confidence_threshold"]'
+        assert_selector 'label[for="chips-fast_context-check"]'
+        assert_selector 'label[for="chips-lemmatize-check"]'
+    end
+
+    test "go to annotator page insert sample text and get annotations" do
+        visit root_url
+        click_link(href: '/annotator')
+
+        # Fill the annotator's text area input by a sample text
+        find(@annotator_text_area).fill_in(with: 'Melanoma is a malignant tumor of melanocytes found mainly')
+
+        # Mock the api call for the annotator with the entered text
+        WebMock.enable!
+        stub_request(:get, "#{@annotator_api}?class_hierarchy_max_level=None&confidence_threshold=0&score_threshold=0&text=Melanoma%20is%20a%20malignant%20tumor%20of%20melanocytes%20found%20mainly&whole_word_only=true")
+            .with(
+                headers: {
+                        'Accept'=>'application/json',
+                        'Authorization'=>"apikey token=#{@apikey}",
+                        'Host'=> @host,
+                        'User-Agent'=>'NCBO API Ruby Client v0.1.0'
+                })
+            .to_return(status: 200, body: @sample_response.to_json, headers: {})
+        
+        
+        find(".annotator-page-button #annotator").click
+
+        # Check if we get the table of annotations
+        assert_selector 'table#annotator-table'
+
+        # Check if the number of annotations is 4
+        assert_equal 5, page.all('tr').count
```

Comment:
assert that the content is good and not just 5 line that are empty or with the wrong content.

Reply:
Okay